traits/src/registry.rs

```diff
@@ -76,3 +77,81 @@ pub trait AccountIdFor<Assets> {
 	/// Create a name to uniquely identify a share token for given assets and an identifier.
 	fn name(assets: &Assets, identifier: Option<&[u8]>) -> Vec<u8>;
 }
+
+use frame_support::dispatch::Parameter;
+
+pub trait Inspect<AssetNativeLocation, Balance> {
+	type Error;
+	type AssetId: Parameter;
+
+	fn is_sufficient(id: Self::AssetId) -> bool;
+}
+
+#[allow(clippy::too_many_arguments)]
+pub trait Create<AssetNativeLocation, Balance>: Inspect<AssetNativeLocation, Balance> {
+	fn register_asset(
+		asset_id: Option<Self::AssetId>,
+		name: Option<&[u8]>,
+		kind: AssetKind,
+		existential_deposit: Option<Balance>,
+		symbol: Option<&[u8]>,
+		decimals: Option<u8>,
+		location: Option<AssetNativeLocation>,
+		xcm_rate_limit: Option<Balance>,
+		is_sufficient: bool,
+	) -> Result<Self::AssetId, Self::Error>;
+
+	fn register_insufficient_asset(
```

Comment:
Why need of special insufficient and sufficient functions ? 

isnt the register_asset enough ?

Reply:
`register_asset` is enough but sufficiency is important (dest. account has to pay ED for insufficient assets on deposit/transfer).
I think it help with readability and can ease debugging/review when 3-rd person look at the code